Anexia: allow extended disk configuration in AnexiaNodeSpec

LittleFox94 opened the pull request:
**What this PR does / why we need it**:

This extends `apiv1.AnexiaNodeSpec` to allow extended disk configuration introduced in machine-controller with https://github.com/kubermatic/machine-controller/pull/1402.

Without this, Kubermatic API returns a 500 when trying to do any `MachineDeployment` operation on a cluster which has a `MachineDeployment` with the new `disks` attribute (and thus without the old `diskSize` attribute).


**What type of PR is this?**
/kind bug
/kind feature
/kind api-change

It's a bug (500 error) because of a new feature, which causes and api-change

**Special notes for your reviewer**:

Comments on the machine-controller PRs:
* https://github.com/kubermatic/machine-controller/pull/1405#issuecomment-1220500780
* https://github.com/kubermatic/machine-controller/pull/1402#issuecomment-1220506220

If the [backport to machine-controller v1.45](https://github.com/kubermatic/machine-controller/pull/1405) is approved, we need this backported to kubermatic version v2.19 and v2.20.

**Does this PR introduce a user-facing change? Then add your Release Note here**:
<!--
Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
Extend disk configuration for provider Anexia
```

**Documentation**:
<!--
Please do one of the following options:
- Add a link to the existing documentation
- Add a link to the kubermatic/docs pull request
- If no documentation change is applicable then add:
  - TBD (documentation will be added later)
  - NONE (no documentation needed for this PR)
-->
```documentation

```

LittleFox94 committed:
Anexia: allow extended disk configuration in AnexiaNodeSpec

This extends apiv1.AnexiaNodeSpec to allow extended disk configuration
introduced in machine-controller with pull request 1402
(https://github.com/kubermatic/machine-controller/pull/1402).

Without this, Kubermatic API returns a 500 when trying to do any
MachineDeployment operation on a cluster which has a MachineDeployment
with the new disks attribute (and thus without the old diskSize
attribute).

Signed-off-by: Mara Sophie Grosch <mgrosch@anexia-it.com>

LittleFox94 committed:
Anexia: disable OSM in E2E tests

Signed-off-by: Mara Sophie Grosch <mgrosch@anexia-it.com>

LittleFox94 commented on cmd/conformance-tester/pkg/scenarios/anexia.go:
```diff
@@ -101,7 +101,31 @@ func (s *anexiaScenario) NodeDeployments(_ context.Context, num int, secrets typ
+				},
+			},
+		},
+		{
```
have a hard time reading the prow log, is this doing what I want?

Create an additional machine deployment, this time with the new Disks attribute instead of the old DiskSize attribute. Having both MachineDeployments around, makes sure in E2E testing, that both variants work as expected.

ahmedwaleedmalik replied:
Yeah, that seems right.

LittleFox94 committed:
Anexia: override default usage of ignition in e2e tests

Signed-off-by: Mara Sophie Grosch <mgrosch@anexia-it.com>

LittleFox94 committed:
conformance-tester, scenario Anexia: fix setting cloud-init in OS Spec

Setting the OperatingSystemSpec on the NodeSpec given into
machine.GetAnexiaProviderConfig() does not actually do anything with the
OS spec, instead, when creating the MachineDeployment, another one is
created - completely ignoring us having set the ProvisioningUtility to
CloudInit.

This fixes that problem by copying the relevant code from the
baseScenario.createMachineDeployment method and modifying the OS spec
given to the createMachineDeployment() function.

Signed-off-by: Mara Sophie Grosch <mgrosch@anexia-it.com>

LittleFox94 committed:
conformance-tester/baseScenario: correctly encode OperatingSystemSpec

OperatingSystemSpec directly contains the config attributes for the
given operating system, but was wrong encoded as
{"$operatingSystem": {$OperatingSystemSpec...}}

This fixes that issue by adding another switch-case over all known
operating systems. Should probably be done in a better way, but need
this done already..

Signed-off-by: Mara Sophie Grosch <mgrosch@anexia-it.com>

ahmedwaleedmalik reviewed (approved):
/approve 🎆
